config.go

```diff
@@ -1,14 +1,93 @@
 package main
 
-import "fmt"
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+)
+
+const (
+	MAX_CLOUD_RESOURCE_NAME_LENGTH = 63
+)
+
+type networkEndpointGroupConfiguration struct {
+	Name string `json:"name"`
+}
+
+type backendServiceConfiguration struct {
+	Name     string `json:"name"`
+	CDN      bool   `json:"cdn"`
+	Affinity string `json:"affinity"`
+	Host     string `json:"host"`
+	Path     string `json:"path"`
+}
 
 type healthCheckConfiguration struct {
+	Name string `json:"name"`
 	Type string `json:"type"`
 	Path string `json:"path"`
 }
 
+type mode string
+
+const (
+	REUSE_MODE   = "REUSE"
+	TAG_MODE     = "TAG"
+	DEFAULT_MODE = "DEFAULT"
+)
+
 type tagConfiguration struct {
-	HealthCheck healthCheckConfiguration `json:"healthCheck"`
+	Name                 string                             `json:"name"`
+	Mode                 mode                               `json:"Mode"`
+	NetworkEndpointGroup *networkEndpointGroupConfiguration `json:"networkEndpointGroup"`
+	BackendService       *backendServiceConfiguration       `json:"backendService"`
+	HealthCheck          *healthCheckConfiguration          `json:"healthCheck"`
+}
+
+func (config *tagConfiguration) validate() error {
+	tag := config.Name
+	if tag == "" {
+		return fmt.Errorf("Tag %q is misconfigured (name is not specified)", tag)
+	}
+
+	switch config.Mode {
+	case DEFAULT_MODE:
+		fallthrough
+	case TAG_MODE:
+		switch {
+		case config.HealthCheck == nil:
+			return fmt.Errorf("No health check configuration specified for tag %q", tag)
+		case config.HealthCheck.Name == "":
+			return fmt.Errorf("No health check name specified for tag %q", tag)
+		case config.HealthCheck.Type != "http":
+			return fmt.Errorf("Health check type is not supported %q for tag %q", config.HealthCheck.Type, tag)
+		case config.HealthCheck.Path == "":
+			return fmt.Errorf("No health check path specified for tag %q", tag)
+		case config.BackendService == nil:
+			return fmt.Errorf("No backend service configuration specified for tag %q", tag)
+		case config.BackendService.Name == "":
+			return fmt.Errorf("No backend service name specified for tag %q", tag)
+		case config.BackendService.Affinity != "noaffinity" &&
+			config.BackendService.Affinity != "ipaffinity" &&
+			config.BackendService.Affinity != "ipportaffinity":
+			return fmt.Errorf("Backend service affinity is not supported %q for tag %q", config.BackendService.Affinity, tag)
+		case config.BackendService.Host == "":
+			return fmt.Errorf("No backend service host specified for tag %q", tag)
+		case config.BackendService.Path == "":
+			return fmt.Errorf("No backend service path specified for tag %q", tag)
```

Comment:
What about an explicit `default` case? As a defensive measure.

Reply:
Looks like we managed in Slack. So resolving the conversation.